arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentTableViewController.swift

```diff
@@ -56,6 +58,10 @@ class ContentTableViewController: UITableViewController {
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         selectedSample = nil
+        if isFavoritesCategory {
+            allSamples = allSamples.filter { $0.isFavorite }
+            tableView.reloadData()
+        }
```

Comment:
Do you mean to subclass `ContentTableViewController` as `class FavoritesCategoryViewController: ContentTableViewController`? Because the favorites are displayed in a table view.

Reply:
Yes, although it has been so long since I've used a UITableView that I forgot about the data source. Instead of subclassing `ContentTableViewController`, you use two different data sources (instance of type that conforms to `UITableViewDataSource`--`UITableViewDiffableDataSource` would be a good option): one that's backed by an array of samples and another one that is backed by the favorite samples, with the logic to get the favorites and keep it up-to-date.